alert-common/src/main/java/com/synopsys/integration/alert/common/persistence/util/ConfigurationFieldModelConverter.java

```diff
@@ -95,6 +99,28 @@ public final FieldAccessor convertToFieldAccessor(final FieldModel fieldModel) t
         return configurationModels;
     }
 
+    public Map<String, FieldValueModel> convertToFieldValuesMap(final Collection<ConfigurationFieldModel> configurationFieldModels) {
+        final Map<String, FieldValueModel> fields = new HashMap<>();
+        for (final ConfigurationFieldModel fieldModel : configurationFieldModels) {
+            final String key = fieldModel.getFieldKey();
+            final Collection<String> values = fieldModel.getFieldValues();
+            final FieldValueModel fieldValueModel = new FieldValueModel(values, fieldModel.isSet());
+            fields.put(key, fieldValueModel);
+        }
+        return fields;
+    }
+
+    public FieldModel convertToFieldModel(final ConfigurationModel configurationModel) throws AlertDatabaseConstraintException {
+        final Long configId = configurationModel.getConfigurationId();
+        final String descriptorName = getDescriptorName(configurationModel);
+        final Map<String, FieldValueModel> fields = new HashMap<>();
+        for (final ConfigurationFieldModel fieldModel : configurationModel.getCopyOfFieldList()) {
```

Comment:
Why couldn't you use the convertToFieldValuesMap method here on the copy of field list and then iterate through the map and use the .setValues (null) method call if the isSensitive field returns true?  That way there is one method to handle converting the configuration model to a map of FieldValueModels.

Reply:
It's a good idea, but we actually need the ConfigurationFieldModel to tell us if a field is sensitive or not. Using that flag, we empty the values.